src/MovingAverage.cpp

```diff
@@ -0,0 +1,111 @@
+/*
+ * Moving Average Filter (linear FIR filter with equal weights)
+ */
+#include "iarc7_sensors/MovingAverage.hpp"
+
+#include <ros/ros.h>
+
+namespace iarc7_sensors {
+
+MovingAverage::MovingAverage(const ros::Duration& time) : filter_time_constant_(time) {}
+
+void MovingAverage::updateFilter(double value, double derivative, const ros::Time& time) {
```

Comment:
We don't appear to be using derivative anywhere in this class. I think we should adapt this class to average the value and the derivative at the same time because they use the same timestamps and the calculation will be more efficient than using two averages in altimeter filter. I'm willing to sacrifice that much of the reusability of this class.

This points out that this class really is a timestamp weight handler. It would be interesting to see this class take a generic type that handles multiple data points simultaneously when operations are applied to it. Probably not useful for this time around though.